extensions/parquet/base/src/main/java/io/deephaven/parquet/base/RowGroupReaderImpl.java

```diff
@@ -34,40 +37,74 @@ final class RowGroupReaderImpl implements RowGroupReader {
+        chunkMapByFieldId = new HashMap<>(fieldCount);
+        final Iterator<Type> fieldsIt = schema.getFields().iterator();
+        final Iterator<ColumnChunk> colsIt = rowGroup.getColumnsIterator();
+        while (fieldsIt.hasNext() && colsIt.hasNext()) {
+            final Type ft = fieldsIt.next();
+            final ColumnChunk column = colsIt.next();
+            final List<String> path_in_schema = column.getMeta_data().path_in_schema;
+            final String key = path_in_schema.toString();
+            final List<Type> nonRequiredFields = new ArrayList<>();
             for (int indexInPath = 0; indexInPath < path_in_schema.size(); indexInPath++) {
                 Type fieldType = schema
                         .getType(path_in_schema.subList(0, indexInPath + 1).toArray(new String[0]));
                 if (fieldType.getRepetition() != Type.Repetition.REQUIRED) {
                     nonRequiredFields.add(fieldType);
                 }
             }
+            chunkMap.put(key, column);
             schemaMap.put(key, nonRequiredFields);
+            if (ft.getId() != null) {
+                chunkMapByFieldId.put(ft.getId().intValue(), column);
+                schemaMapByFieldId.put(ft.getId().intValue(), nonRequiredFields);
+            }
+        }
+        if (fieldsIt.hasNext() || colsIt.hasNext()) {
+            throw new IllegalStateException(String.format("Unexpected, iterators not exhausted, rootURI=%s", rootURI));
         }
         this.version = version;
     }
 
     @Override
     @Nullable
-    public ColumnChunkReaderImpl getColumnChunk(@NotNull final String columnName, @NotNull final List<String> path) {
-        final String key = path.toString();
-        final ColumnChunk columnChunk = chunkMap.get(key);
-        final List<Type> fieldTypes = schemaMap.get(key);
-        if (columnChunk == null) {
-            return null;
+    public ColumnChunkReaderImpl getColumnChunk(@NotNull final String columnName, @NotNull final List<String> path,
+            @Nullable Integer fieldId) {
+        final ColumnChunk columnChunk;
+        final List<Type> nonRequiredFields;
+        PARTS: {
+            if (fieldId != null) {
+                final ColumnChunk cc = chunkMapByFieldId.get(fieldId);
+                if (cc != null) {
+                    columnChunk = cc;
+                    nonRequiredFields = schemaMapByFieldId.get(fieldId);
+                    break PARTS;
+                }
+            }
+            final String key = path.toString();
+            final ColumnChunk cc = chunkMap.get(key);
+            if (cc == null) {
+                return null;
+            }
+            columnChunk = cc;
+            nonRequiredFields = schemaMap.get(key);
         }
-        return new ColumnChunkReaderImpl(columnName, columnChunk, channelsProvider, rootURI, type, fieldTypes,
+        return new ColumnChunkReaderImpl(columnName, columnChunk, channelsProvider, rootURI, schema, nonRequiredFields,
```

Comment:
This seems like too much to ask but in the spirit of beautifying this code, can you rename fieldTypes to nonRequiredFields in downstream code as well?